### ISSUE TITLE
Is there a way to have Fetch not create files as soon as downloads are enqueued ?

### ISSUE THREAD

**Issue by thib-rdr** [tonyofrancis/Fetch#353]
Hi,

Love your lib, very powerful.

I'm using it to sync a large number of files between a client and a server. Each time a user hits the sync button, I get all my files urls from a db in my server, compare them to what I have on my device, and enqueue the missing files to fetch accordingly.

Problem, is that I didn't see that Fetch is creating empty 0b files on the device as soon as downloads are enqueued. So, my comparison between the files in my db and on disk is messed up, as the files are, in fact, already there.
Is there a way for those files not to be created on download queuing, but on download starting ? If yes, that would make my life a lot easier. If not, I'll find a workaround.

Also, out of pure curiosity, what was the reason that led you to create files on enqueue ? Is there a use case I'm not seeing ? 

Thanks very much for your work and your answers.
Thib

**Comment by tonyofrancis** [comment]
@thib-rdr I have to have a look at the enqueue code again to give you better answer. But I do know that it is needed because of the different enqueue settings eg `EnqueueAction` and helping maintain better download status especially when there are several instance of Fetch running with the same namespace etc.

**Comment by thib-rdr** [comment]
Ok, I get it. But IMO, this could be a setting, especially when using `EnqueueAction .UPDATE_ACCORDINGLY`... After looking at the code, I guess it could be feasible...

**Comment by tonyofrancis** [comment]
@thib-rdr I had a look at your PR but It was not the best way to go about this. Just released Fetch version 3.0.4 that handles this correctly. All you have to do is set two options on your Fetch Configuration.
eg: 
```java
        final FetchConfiguration fetchConfiguration = new FetchConfiguration.Builder(this)
                .enableFileExistChecks(false)
                .createDownloadFileOnEnqueue(false)
                .build();
```
Let me know if this works well for you.

**Comment by thib-rdr** [comment]
Awesome ! Thanks very much for your help, that will work for me !
